Make.rules

```diff
@@ -36,19 +36,27 @@
 
 .SECONDARY:
 
+CHAR_SUBSYSTEM=$(subst 0x,\x,$(SUBSYSTEM))
+
 ifeq ($(IS_MINGW32),)
 %.efi: %.so
 	@$(ECHO) "  OBJCOPY  $(notdir $@)"
 	$(HIDE)$(OBJCOPY) -j .text -j .sdata -j .data -j .dynamic -j .rodata -j .rel \
 		    -j .rela -j .rel.* -j .rela.* -j .rel* -j .rela* \
 		    -j .areloc -j .reloc $(FORMAT) $*.so $@
+ifeq ($(ARCH),riscv64)
+	$(HIDE)/bin/echo -ne "$(CHAR_SUBSYSTEM)" | dd of=$@ bs=1 seek=156 count=1 conv=notrunc status=none
+endif
 
 %.efi.debug: %.so
 	@$(ECHO) "  OBJCOPY  $(notdir $@)"
 	$(HIDE)$(OBJCOPY) -j .debug_info -j .debug_abbrev -j .debug_aranges \
 		-j .debug_line -j .debug_str -j .debug_ranges \
 		-j .note.gnu.build-id \
 		$(FORMAT) $*.so $@
+ifeq ($(ARCH),riscv64)
```

Comment:
debug files don't contain PE header

Reply:
OK, Ill fix that. I actually have no idea what these `.efi.debug` files are for...

Reply:
> OK, Ill fix that. I actually have no idea what these `.efi.debug` files are for...

GDB - not that I've ever tried... The process to do it looks complicated
The ctors broke I disassembled instead cos it seemed easier than figuring it out